pkg/model/log.go

```diff
@@ -0,0 +1,172 @@
+	EncodeName       string `json:"nameEncoder" yaml:"nameEncoder"`
+	ConsoleSeparator string `json:"consoleSeparator" yaml:"consoleSeparator"`
+}
+
+func (e *EncoderConfig) build() zapcore.EncoderConfig {
+	return zapcore.EncoderConfig{
+		MessageKey:       e.MessageKey,
+		LevelKey:         e.LevelKey,
+		TimeKey:          e.TimeKey,
+		NameKey:          e.NameKey,
+		CallerKey:        e.CallerKey,
+		FunctionKey:      e.FunctionKey,
+		StacktraceKey:    e.StacktraceKey,
+		SkipLineEnding:   e.SkipLineEnding,
+		LineEnding:       e.LineEnding,
+		EncodeLevel:      e.unmarshalLevelEncoder(),
+		EncodeTime:       e.unmarshalTimeEncoder(),
+		EncodeDuration:   e.unmarshalDurationEncoder(),
+		EncodeCaller:     e.unmarshalCallerEncoder(),
+		EncodeName:       e.unmarshalNameEncoder(),
+		ConsoleSeparator: e.ConsoleSeparator,
+	}
+}
+
+func (e *EncoderConfig) unmarshalLevelEncoder() zapcore.LevelEncoder {
+	switch e.EncodeLevel {
+	case "capital":
+		return zapcore.CapitalLevelEncoder
+	case "capitalColor":
+		return zapcore.CapitalColorLevelEncoder
+	case "color":
+		return zapcore.LowercaseColorLevelEncoder
+	default:
+		return zapcore.LowercaseLevelEncoder
+	}
+}
+
+func (e *EncoderConfig) unmarshalTimeEncoder() zapcore.TimeEncoder {
+	switch e.EncodeTime {
+	case "rfc3339nano", "RFC3339Nano":
+		return zapcore.RFC3339NanoTimeEncoder
+	case "rfc3339", "RFC3339":
+		return zapcore.RFC3339TimeEncoder
+	case "iso8601", "ISO8601":
+		return zapcore.ISO8601TimeEncoder
+	case "millis":
+		return zapcore.EpochMillisTimeEncoder
+	case "nanos":
+		return zapcore.EpochNanosTimeEncoder
+	default:
+		return zapcore.EpochTimeEncoder
+	}
+}
+
+func (e *EncoderConfig) unmarshalDurationEncoder() zapcore.DurationEncoder {
+	switch e.EncodeDuration {
+	case "string":
+		return zapcore.StringDurationEncoder
+	case "nanos":
+		return zapcore.NanosDurationEncoder
+	case "ms":
+		return zapcore.MillisDurationEncoder
+	default:
+		return zapcore.SecondsDurationEncoder
+	}
+}
+
+func (e *EncoderConfig) unmarshalCallerEncoder() zapcore.CallerEncoder {
+	switch e.EncodeCaller {
+	case "full":
+		return zapcore.FullCallerEncoder
+	default:
+		return zapcore.ShortCallerEncoder
+	}
+}
+
+func (e *EncoderConfig) unmarshalNameEncoder() zapcore.NameEncoder {
+	switch e.EncodeName {
+	case "full":
+		return zapcore.FullNameEncoder
+	default:
+		return zapcore.FullNameEncoder
+	}
+}
+
+func (l *Log) Build() *zap.Config {
+	lvl, err := zap.ParseAtomicLevel(l.Level)
+	if err != nil {
+		logger.Errorf("failed parse %s to zap.AtomicLevel", l.Level)
+	}
+
+	return &zap.Config{
+		Level:             lvl,
+		Development:       l.Development,
+		DisableCaller:     l.DisableCaller,
+		DisableStacktrace: l.DisableStacktrace,
+		Sampling:          l.Sampling.build(),
+		Encoding:          l.Encoding,
+		EncoderConfig:     l.EncoderConfig.build(),
+		OutputPaths:       l.OutputPaths,
+		ErrorOutputPaths:  l.ErrorOutputPaths,
+		InitialFields:     l.InitialFields,
+	}
+}
```

Comment:
put this function to below the `Log` definition, just like the other two structures and their corresponding methods.